src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs

```diff
@@ -590,14 +600,35 @@ public bool TryGetRunningSubscriptionConnectionsState(long subscriptionId, out S
             if (subscriptionBlittable == null)
                 return null;
 
-            var subscriptionState = JsonDeserializationClient.SubscriptionState(subscriptionBlittable);
+            if (subscriptionBlittable.TryGet(nameof(SubscriptionState.SubscriptionId), out long id) == false)
+            {
+                if (_logger.IsOperationsEnabled)
+                    _logger.Info($"Could not figure out the Subscription Task ID for subscription named: '{subscriptionName}'.");
 
-            if (_subscriptions.TryGetValue(subscriptionState.SubscriptionId, out SubscriptionConnectionsState concurrentSubscription) == false)
+                return null;
+            }
+
+            if (_subscriptions.TryGetValue(id, out SubscriptionConnectionsState concurrentSubscription) == false)
                 return null;
 
             return concurrentSubscription;
         }
 
+        public class SubscriptionStateWithRaftIndex : SubscriptionState
```

Comment:
why do you need this class?
can't we simply add the `RaftIndex` to the existing `SubscriptionState` ?

Reply:
`SubscriptionState` is part of the Client API, I didn't want to add server stuff to it

Reply:
I think the changes due to the new class are not worth it in this case.
I don't see any harm in adding an additional property to `SubscriptionState`

Reply:
done